rest_api/aggregation/database_hooks.py

```diff
@@ -121,6 +123,42 @@ def _expression(self, species_contam):
         return sorted(k for k, v in species_contam['contaminant_unique_mapped'].items() if v > 500)
 
 
+class RequiredYield(Calculation):
+    @cached_property
+    def quantised_yields(self):
+        return cfg['available_yields']
+
+    def _expression(self, genome_size):
+        return {
+            # keys have to be strings to be valid BSON
+            str(required_coverage) + 'X': self._get_yield(genome_size, required_coverage)
+            for required_coverage in cfg['available_coverages']
+        }
+
+    def _get_yield(self, genome_size, required_coverage):
+        """
+        Obtain a required yield, given a particular genome and required X coverage, using the keys of the
+        'available_yields' config for candidate yield values.
+        :param float genome_size: Genome size in Mb
+        :param int required_coverage: X coverage
+        :return: required yield in Gb
+        """
+        exact_yield = genome_size / 1000 * required_coverage
+        for y in sorted(self.quantised_yields):
+            if y >= exact_yield:
+                return y
+
+
+class RequiredYieldQ30(RequiredYield):
+    def _get_yield(self, genome_size, required_coverage):
+        """
+        Obtain a required yield as per RequiredYield._get_yield, and use the 'available_yields' config to translate it
+        into a required yield q30.
+        """
+        required_yield = super()._get_yield(genome_size, required_coverage)
+        return self.quantised_yields[required_yield]
```

Comment:
This will raise a KeyError if the required_yield is `None`
```suggestion
        if required_yield:
            return self.quantised_yields.get(required_yield)
```